server/app/views/components/Modal.java

```diff
@@ -76,6 +77,12 @@ private DivTag getModalHeader() {
                 .withClasses(ReferenceClasses.MODAL_CLOSE, BaseStyles.MODAL_CLOSE_BUTTON));
   }
 
+  public static String randomModalId() {
+    // We prepend a "a-" since element IDs must start with an alphabetic character, whereas UUIDs
+    // can start with a numeric character.
+    return "a-" + UUID.randomUUID().toString();
```

Comment:
suggestion (non-blocking): might reduce a couple milliseconds of thinking for someone reading the rendered result by prepending `uuid-` instead